openmc/mgxs_library.py

```diff
@@ -1634,6 +1685,267 @@ def set_inverse_velocity_mgxs(self, inverse_velocity, temperature=294.,
         self._inverse_velocity[i] = inverse_velocity.get_xs(
             nuclides=nuclide, xs_type=xs_type, subdomains=subdomain)
 
+    def convert_representation(self, target_representation, num_polar=None,
+                               num_azimuthal=None):
+        """Produce a new XSdata object with the same data, but converted to the
+        new representation
+
+        Parameters
+        ----------
+        target_representation : {'isotropic', 'angle'}
+            Representation of the MGXS (isotropic or angle-dependent flux
+            weighting).
+        num_polar : int, optional
+            Number of equal width angular bins that the polar angular
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+        num_azimuthal : int, optional
+            Number of equal width angular bins that the azimuthal angular
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+
+        Returns
+        -------
+        openmc.XSdata
+            Multi-group cross section data with the same data as self, but
+            represented as specified in :param:`target_representation`.
+
+        """
+
+        check_value('target_representation', target_representation,
+                    _REPRESENTATIONS)
+        if target_representation == 'angle':
+            check_type('num_polar', num_polar, Integral)
+            check_type('num_azimuthal', num_azimuthal, Integral)
+            check_greater_than('num_polar', num_polar, 0)
+            check_greater_than('num_azimuthal', num_azimuthal, 0)
+
+        xsdata = copy.deepcopy(self)
+
+        # First handle the case where the current and requested
+        # representations are the same
+        if target_representation == self.representation:
+            # Check to make sure the num_polar and num_azimuthal values match
+            if target_representation == 'angle':
+                if num_polar != self.num_polar or num_azimuthal != self.num_azimuthal:
+                    raise NotImplementedError("XCannot translate between "
```

Comment:
I dont personally have plans to do this, but there is no reason one could not do it by assuming a histogram representation and using that to convert to a new grid.

Either way, I'm fine with `ValueError`

- [x]